assets/app/view/game/scrap_trains.rb

```diff
@@ -6,14 +6,36 @@ module View
   module Game
     class ScrapTrains < Snabberb::Component
       include Actionable
+      needs :corporation, default: nil
 
       def render
-        @corporation = @game.round.active_step.current_entity
-        step = @game.round.active_step
+        @corporation ||= @game.round.active_step.current_entity
+        step = @game.round.step_for(@corporation, 'scrap_train')
 
         scrappable_trains = step.scrappable_trains(@corporation)
         return nil if scrappable_trains.empty?
 
+        if step.scrap_trains_button_only?
```

Comment:
Can you use `step.respond_to?(:scrap_trains_button_only?) && step.scrap_trains_button_only?`, so the other games aren't required to implement the method?

Then you can remove the method from all the other games and not require future games to implement it either.

Reply:
I've changed this and the related files